3911 do not render empty lists on article page

StephDriver opened the pull request:
Updated django templates to check that there are galleys before creating download and checksum lists, across all three themes. Material already checked for galleys for download lists, but not for checksums.

For empty autogenerated tables of contents in OLH theme, updated the JS to remove the `toc` element if no headings were found, i.e. if list not populated.

Marked as draft as it is the end of the day, and I have yet to test this, so that must be done before sending for review.

StephDriver commented:
### Download and Checksum were addressed on the template.

Before:
<img width="347" alt="image" src="https://github.com/BirkbeckCTP/janeway/assets/5330770/6081b914-b54c-46b2-9b50-388ff7b3c497">

```html
<div class="section">
   <h3>Download</h3>
    <ul> 
    </ul>
</div>
```
<img width="323" alt="image" src="https://github.com/BirkbeckCTP/janeway/assets/5330770/3534076f-9aae-4452-82b3-c96c38bb964f">

After:

```html
<div class="section">                             
    <h3>Download</h3>
     <p> Downloads are not available for this article.</p>
</div>
```

### Autogenerated TOC addressed in the JS
This removes the "jump to" section completely.

StephDriver commented:
### To Remove Empty Sections or Not to Remove...
For Checksum and Download, I have left the section there but added a statement that there are none available.  This is because these are standard sections that a reader might 'look for' and so they should not be hidden but state clearly that they are empty.

But for the automated table of contents, I have removed the section entirely when it is empty.  This is because the table of contents listing the sections does not make any sense in the context of an article without contents - and a reader would not look for it when there is no contents to navigate. To follow the pattern above with downloads and checksums would add to user confusion not prevent it.

StephDriver commented:
Closes #3911

joemull commented on src/themes/OLH/templates/journal/article.html:
```diff
@@ -351,19 +351,24 @@ <h4>{% trans "Authors" %}</h4>
+
                                     {% if proofing %}
                                         <p id="note_to_proofreader_2">
                                             <i class="fa fa-info"></i>
```
This note to the proofreader should show up only if there are galleys, as otherwise it may be taken to mean that there may be galleys that are not showing, with the new "Downloads are not available" message. Could we put it inside `if galleys` above?

StephDriver replied:
done

joemull reviewed (changes requested):
Nice one! I like the design decision to consistently display sections that are standard and often expected, like downloads and checksums.

I noticed a few things that we probably want to fix up for consistency between themes and screen sizes.

StephDriver commented on src/themes/clean/templates/journal/article.html:
```diff
@@ -100,15 +100,19 @@ <h2>{% trans "Abstract" %}</h2>
                     <div class="d-lg-none d-xl-none d-md-none">
```
yes, ooops missed that!

StephDriver commented on src/themes/material/templates/journal/article.html:
```diff
@@ -254,13 +254,13 @@ <h4>
                         <a class="waves-effect waves-light btn btn-small linkedin-bg"
                             href="https://www.linkedin.com/sharing/share-offsite?url={% if doi_url %}{{ doi_url }}{% else %}{{ article.url }}{% endif %}"
-                            target="_blank"><i class="fa fa-linkedin"></i></a></button>
+                            target="_blank" aria-label={% trans "Share on Linked In" %}><i class="fa fa-linkedin"></i></a></button>
```
This has made me really think - the aria-label is to be read aloud, so I was writing it as it would be said, so I've had to check weather a screen reader will read LinkedIn intelligibly - and you are right, it reads it fine!

StephDriver commented on src/themes/material/templates/journal/article.html:
```diff
@@ -473,13 +473,17 @@ <h4>{% trans field.field.name %}</h4>
                         <h4>
                             {% trans "File Checksums" %} (MD5)
                         </h4>
```
I'm searching each of the article files for Download and Checksum areas and ensuring consistency of this behaviour across all of them.

NB as far as I can tell, we only show Checksums at large screen-width.  Is that intentional?

StephDriver commented:
Summary of Changes:

- I have made the changes requested.
- I retested the new functionality (as described in previous comment) and updated where needed.
- I found that the Material and Clean themes use the same toc.js asset, which is in the Material directory, this seems to be a carbon copy of the toc.js in the OLH assets.  I updated that toc.js and added the 'toc-section' id to the article.html files so that the toc-section can be hidden when no headings have been found while generating the toc. 
- The material theme tested fine, because it has other functionality which suppresses the TOC for short articles, such as the test example. The TOC card is not visible until one scrolls further down the page, and the page is too short to reach that point which is why it appeared to work correctly before the material toc.js update.  I decided that for consistency, I would update the material article.html to be consistent with the other two themes, despite this... which means if later updates are made to the material theme which change when the TOC card is visible, such that it is visible on shorter pages, the purpose of this issue PR, to hide the TOC when there is no content for it, will then function consistently with the other two themes.   But I am unable to directly test it working - I can note that it is exactly the same as in the Clean theme and that does work.

joemull commented on src/themes/clean/templates/journal/article.html:
```diff
@@ -118,7 +121,6 @@ <h2>{% trans "Abstract" %}</h2>
                         {% endif %}
```
This is no longer inside the if galleys check.

StephDriver replied:
ta, its been a bit of a monster to find all the relevant bits, so I'm glad you spotted that.

joemull reviewed (changes requested):
Amazingly thorough! I just noticed one proofreader note that got left out of one of the galley checks.

ajrbyers reviewed (changes requested):
Thanks for this @StephDriver this looks awesome. However, there is a conflict being flagged. Looks like an easy one though. If you need assistance with it let me know.
